### DISCUSSION TITLE
Puppeteer问题

### DISCUSSION THREAD

**Discussion by A2Sumie** [Tsuk1ko/CQ-picfinder-robot#383] [问题求助] [answered]
NodeJS 16，debian 10（ARM），今天新部署的,3.1.2
有没有什么思路

ascii2d 搜索失败
Could not find Chromium (rev. 1069273). This can occur if either
 1. you did not perform an installation before running the script (e.g. `npm install`) or
 2. your cache path is incorrectly configured (which is: /home/pi/.cache/puppeteer).
For (2), check out our guide on configuring puppeteer at https://pptr.dev/guides/configuration.

其中他给出的文件夹里面是一个空的chrome文件夹

**Comment by Tsuk1ko** [comment, marked as answer]
感觉是 puppeteer 没有提供 arm 的 chromium

https://github.com/puppeteer/puppeteer/issues/7740

可能有一个方法是你自行安装 chromium，然后修改这里的 `executablePath` 提供可执行文件的路径，如果可行的话我可以增加一个自定义 `executablePath` 的配置

https://github.com/Tsuk1ko/cq-picsearcher-bot/blob/b309fd2484028e27d7e910e86ac3644f970421e7/libs/puppeteer/index.mjs#L11-L15

**Reply by A2Sumie** [reply]
遇到Mrs4s/go-cqhttp/issues/1795，在重新看，重新弄一下真是坏事扎堆了

**Reply by Tsuk1ko** [reply]
那可能得等 go-cqhttp 适配了，看 https://github.com/takayama-lily/oicq/pull/475 没法从现有错误链接转换成对的链接，取值都不一样

**Reply by A2Sumie** [reply]
换了个手机试，很久没输出没报错，查进程能看见chromium启动了，我这平台上的chromium也是古物说不清了
